src/led/libled.h

```diff
@@ -0,0 +1,429 @@
+/*
+ * LED library public interface
+ *
+ * Copyright (C) 2022-2022 Red Hat, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef _LIB_LED_INCLUDED_
+#define _LIB_LED_INCLUDED_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <linux/limits.h>
+#include <stdbool.h>
+
+#define LED_SYM_PUBLIC __attribute__((visibility("default")))
+
+typedef int led_status_t;
+
+struct led_ctx;
+
+enum led_log_level_enum {
+	LED_LOG_LEVEL_UNDEF = 0,
+	LED_LOG_LEVEL_QUIET,
+	LED_LOG_LEVEL_ERROR,
+	LED_LOG_LEVEL_WARNING,
+	LED_LOG_LEVEL_INFO,
+	LED_LOG_LEVEL_DEBUG,
+	LED_LOG_LEVEL_ALL,
+};
+
+enum led_cntrl_type {
+	LED_CNTRL_TYPE_UNKNOWN = 0,
+	LED_CNTRL_TYPE_DELLSSD,
+	LED_CNTRL_TYPE_VMD,
+	LED_CNTRL_TYPE_SCSI,
+	LED_CNTRL_TYPE_AHCI,
+	LED_CNTRL_TYPE_NPEM,
+	LED_CNTRL_TYPE_AMD,
+};
+
+/**
+ * @brief The following functions are for functionality pertaining to hardware controllers.
+ */
+
+/**
+ * @brief Translates a controller string ID to enumerated type
+ * 
```

Comment:
> I strongly recommend you to use "Highligh Trailing White Spaces" extension in you IDE.

I originally was auto stripping WS on file save, but that introduced non-relevant changes because the existing source has trailing WS in it.

Reply:
Done